specs/version_format_specification.md

```diff
@@ -0,0 +1,87 @@
+---
+title: Version Format Specification
+layout: default
+version: 1
+SPDX-License-Identifier: CC-BY-4.0
+---
+
+# Version Format Specification
+Many of the formats and concepts defined in these specifications rely on having
+a sort order for strings that include version components, and use it for various
+purposes, such as choosing the default boot entry in the
+[Boot Loader Specification](boot_loader_specification.md). This specification
+defines what format these version strings must follow, and how to
+deterministically compare them.
```

Comment:
This is still not ideal, because it suggests that the comparison is defined for the version component, when in fact it's defined for arbitrary strings. I can compare `systemd-253-6.fc38.i686` with `systemd-253-6.fc38.i686~` and I get a reasonable and deterministic result. This is a very important aspect of this algorithm and how it's used in practice.

Reply:
dropped `version`

Reply:
Maybe separate the second commit to a second PR? I'd like to get the first part merged and start adjusting URLs in other places and such…